conn.go

```diff
@@ -124,11 +130,25 @@ func (c *conn) QueryContext(ctx context.Context, query string, args []driver.Nam
 		return nil, err
 	}
 
-	return &rows{ctx: ctx, conn: c, currentRowIdx: -1, currentPartitionIdx: 0, currentResultSet: rs}, nil
+	if rs.Metadata.DataplaneRequest != nil {
+		if rs.Metadata.DataplaneRequest.RequestType == apiv2.DataplaneRequestRequestTypeResultSet {
+			dpconn, err := NewDPConn(*rs.Metadata.DataplaneRequest, c.sessionID, c.httpClient)
```

Comment:
Does this conn need to be closed?

Reply:
no, it is just holding a reference to the dp request. it doesnt hold anything open.